File: serverapp/src/app/interface/student.ts
import { Levels } from '../enum/levels.enum';

export interface Student {
  id?: number;
  name: string;
  isPreply: boolean;
  level: Levels;
  progressInfo: string;
  objectivesInfo: string;
  nextClassInfo: string;
  hobbiesInfo: string;
  numPaidClasses: number;
}

File: server/src/main/java/io/list/server/model/Student.java
package io.list.server.model;

import io.list.server.enumeration.Levels;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;


import static javax.persistence.GenerationType.AUTO;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Student {
  @Id
  @GeneratedValue(strategy = AUTO)
  private Long id;

  private String name;
  private Boolean isPreply;
  private Levels level;
  private String progressInfo;
  private String objectivesInfo;
  private String nextClassInfo;
  private String hobbiesInfo;
  private Integer numPaidClasses;
}



